### DISCUSSION TITLE
Debugging CSharp Transform Scripts

### DISCUSSION THREAD

**Discussion by ThetaS** [bonsai-rx/bonsai#974] [Q&A] [answered]
Hi Bonsai team,

I am curious about how can I debug the C# program associated to one of the transforms used in my workflow if I use VS Code as my code editor.  I read the instructions here: https://bonsai-rx.org/docs/create-package/ , but it seems like this method does not work with VS Code. May I have some advice on that? Thanks.

**Comment by bruno-f-cruz** [comment, marked as answer]
You wont be able to have breakpoints in VSCode. There are several reasons for this, one of them being that extension only gets compiled by bonsai when the latter reloads.

However, something super useful when scripting in VSCode is autocomplete. Make sure to install [NET Framework 4.7.2 Developer Pack](https://dotnet.microsoft.com/download/dotnet-framework/thank-you/net472-developer-pack-offline-installer) to enable it.
